controller/api/destination/endpoint_translator.go

```diff
@@ -310,6 +478,33 @@ func (et *endpointTranslator) sendClientRemove(set watcher.AddressSet) {
 	}
 }
 
+// closeEndpointPolicy closes policy server watches for pod addresses that
+// have been removed. If there is a policy server client for an address in the
+// set, it is closed and then removed from the policyWatches map.
+func (et *endpointTranslator) closeEndpointPolicy(set watcher.AddressSet) {
+	// When deleting from the the policy watches map, ensure clients are not
+	// also being added by watchEndpointPolicy
+	et.policyWatches.mutex.Lock()
+	defer et.policyWatches.mutex.Unlock()
+
+	for _, addr := range set.Addresses {
+		// If the address is not backed by a pod then there will be no policy
+		// server watch to close.
+		if addr.Pod == nil {
+			continue
+		}
+		portSpec := portSpec{
+			workload: fmt.Sprintf("%s:%s", addr.Pod.Namespace, addr.Pod.Name),
+			port:     addr.Port,
+		}
+		if portClient, ok := et.policyWatches.watches[portSpec]; ok {
+			et.log.Debugf("Closing policy server client for %s:%d", portSpec.workload, portSpec.port)
+			portClient.cancel()
+		}
+		delete(et.policyWatches.watches, portSpec)
```

Comment:
Similar to by comment above: what if there are multiple destination requests open which include the same pod+port?  If the pod is removed from one service's endpoints but remains in a second service's endpoints, I think this would close the policy watch.